frontend/packages/console-shared/src/hooks/select-list.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { OnSelect } from '@patternfly/react-table';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+const getAllSelectedRows = <R extends K8sResourceCommon>(data: R[], selectedRows: string[]) =>
+  data.filter((row) => selectedRows.includes(row.metadata.uid));
+
+export const useSelectList = <R extends K8sResourceCommon>(
+  data: R[],
+  visibleRows: string[],
+  onRowSelected: (rows: R[]) => void,
+): {
+  onSelect: OnSelect;
+  selectedRows: string[];
+  updateSelectedRows: (rows: R[]) => void;
+} => {
+  const [selectedRows, setSelectedRows] = React.useState<string[]>([]);
+
+  const onSelect = (_event, isSelected, rowIndex, rowData) => {
+    let uniqueUIDs = [];
+    if (rowIndex === -1) {
```

Comment:
when can rowIndex be -1 ?

I dont understand why this is not just 

```
const selectedUIDs = isSelected ? selectedRows.push(rowData.props.id) : selectedRows.filter((uid) => rowData.props.id != uid);
setSelectedRows(selectedUIDs);
```

why do we need to do all the uniq and xors ?

Reply:
When you click select all on top of the table you will get a rowIndex of -1.

Reply:
Yes, when you select checkbox that corresponds to select all nodes, the rowIndex comes as -1.

Reply:
uniq is done to uniquely get the uid of the nodes as its possible that the `visibleRows` and `selectedRows` can have common uid in it.